proposal.rst

```diff
@@ -152,14 +150,17 @@ Motivations
 
 **Why have you chosen this particular project?**
 
-FILLME
+I have read several articles about how Google search works and it is get interesting to me to have some experience in this area.  
 
 **Who will benefit from your project and in what ways?**
 
 .. For example, think about the likely user-base, what they currently have to
 .. do and how your project will improve things for them.
 
-FILLME
+This project aimed at search engines developers, it could provide kit of ranking algorithms with ability to change dataset automatically.
+I do not have full understandind Xapian structure now, but I think it could be two ways of user interaction with my part of project:
+	1) Letor and Omega as a parts of Xapian will provide set of user functions to Xapian API. So user will have ability to collect data by Omega and rank dataset by one of Letor algorithms.
+	2) Create separate application for managing collecting and ranking process.  
```

Comment:
While you don't need to fully understand Xapian's structure in every detail (not even to complete the project as there are parts of Xapian you just don't need to know about), you really do need to have at least a high level understanding of the structure of the components that your project is directly involved with in order to propose a meaningful project plan.

For example, Omega is a set of programs (under xapian-applications/omega in the source tree) which make use of the Xapian C++ API (under xapian-core) - there isn't really any way Omega can add functions to that API.